credential-exchange-types/src/format/editable_field.rs

```diff
@@ -0,0 +1,362 @@
+        })
+    }
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq, Eq)]
+#[serde(transparent)]
+pub struct EditableFieldString(String);
+impl EditableFieldType for EditableFieldString {
+    fn field_type(&self) -> FieldType {
+        FieldType::String
+    }
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq, Eq)]
+#[serde(transparent)]
+pub struct EditableFieldConcealedString(String);
+impl EditableFieldType for EditableFieldConcealedString {
+    fn field_type(&self) -> FieldType {
+        FieldType::ConcealedString
+    }
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct EditableFieldBoolean(
+    #[serde(
+        serialize_with = "serialize_bool",
+        deserialize_with = "deserialize_bool"
+    )]
+    bool,
+);
+impl EditableFieldType for EditableFieldBoolean {
+    fn field_type(&self) -> FieldType {
+        FieldType::Boolean
+    }
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq, Eq)]
+#[serde(transparent)]
+pub struct EditableFieldDate(String);
+impl EditableFieldType for EditableFieldDate {
+    fn field_type(&self) -> FieldType {
+        FieldType::Date
+    }
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq, Eq)]
+#[serde(transparent)]
+pub struct EditableFieldYearMonth(String);
+impl EditableFieldType for EditableFieldYearMonth {
+    fn field_type(&self) -> FieldType {
+        FieldType::YearMonth
+    }
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq, Eq)]
+#[serde(transparent)]
+pub struct EditableFieldSubdivisionCode(String);
+impl EditableFieldType for EditableFieldSubdivisionCode {
+    fn field_type(&self) -> FieldType {
+        FieldType::SubdivisionCode
+    }
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq, Eq)]
+#[serde(transparent)]
+pub struct EditableFieldCountryCode(String);
+impl EditableFieldType for EditableFieldCountryCode {
+    fn field_type(&self) -> FieldType {
+        FieldType::CountryCode
+    }
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq, Eq)]
+#[serde(rename_all = "kebab-case")]
+pub enum EditableFieldWifiNetworkSecurityType {
+    Unsecured,
+    WpaPersonal,
+    Wpa2Personal,
+    Wpa3Personal,
+    Wep,
+
+    #[serde(untagged)]
+    Other(String),
+}
+impl EditableFieldType for EditableFieldWifiNetworkSecurityType {
+    fn field_type(&self) -> FieldType {
+        FieldType::WifiNetworkSecurityType
+    }
+}
+
+fn serialize_bool<S>(value: &bool, serializer: S) -> Result<S::Ok, S::Error>
+where
+    S: serde::Serializer,
+{
+    serializer.serialize_str(if *value { "true" } else { "false" })
+}
+
+fn deserialize_bool<'de, D>(deserializer: D) -> Result<bool, D::Error>
+where
+    D: serde::Deserializer<'de>,
+{
+    let value = <&str>::deserialize(deserializer)?;
+    match value.trim().to_lowercase().as_str() {
+        "true" => Ok(true),
+        "false" => Ok(false),
+        _ => Err(serde::de::Error::custom("expected 'true' or 'false'")),
+    }
+}
```

Comment:
Question, why not use `to_string()` and `value.parse()` instead?
```suggestion
fn serialize_bool<S>(value: &bool, serializer: S) -> Result<S::Ok, S::Error>
where
    S: serde::Serializer,
{
    serializer.serialize_str(&value.to_string())
}

fn deserialize_bool<'de, D>(deserializer: D) -> Result<bool, D::Error>
where
    D: serde::Deserializer<'de>,
{
    let value = <&str>::deserialize(deserializer)?;
    value
        .trim()
        .to_lowercase()
        .parse()
        .map_err(serde::de::Error::custom)
}
```

Reply:
Mostly because I wasn't aware of that functionality.